src/mg/PAMI/Client/Impl/ClientImpl.php

```diff
@@ -405,20 +405,21 @@ public function send(OutgoingMessage $message)
 	    if (@fwrite($this->_socket, $messageToSend) < $length) {
     	    throw new ClientException('Could not send message');
 	    }
-	    $read = 0;
-	    while($read <= $this->_rTimeout) {
-	        $this->process();
-	        $response = $this->getRelated($message);
-	        if ($response != false) {
-	            $this->_lastActionId = false;
-	            return $response;
-	        }
-	        usleep(1000); // 1ms delay
-	        if ($this->_rTimeout > 0) {
-	            $read++;
-	        }
-	    }
-	    throw new ClientException('Read timeout');
+	    while (1) {
+	    	stream_set_timeout($this->_socket, $this->_rTimeout ? $this->_rTimeout : 1);
```

Comment:
Actually stream_socket_client seems to be tracking cTimeout already, so that should be good enough ! It will mean that we need to check error returns from stream_set_timeout and stream_get_meta_data.